cpp/src/parquet/arrow/writer.cc

```diff
@@ -403,17 +410,31 @@ class FileWriterImpl : public FileWriter {
     }
 
     auto WriteBatch = [&](int64_t offset, int64_t size) {
+      std::vector<std::unique_ptr<ArrowColumnWriterV2>> writers;
+
       int column_index_start = 0;
       for (int i = 0; i < batch.num_columns(); i++) {
-        ChunkedArray chunkedArray(batch.column(i));
+        ChunkedArray chunkedArray{batch.column(i)};
```

Comment:
Here would `chunkedArray` dtor in stack because it outlives writer's lifetime?

Reply:
If the optimization is aggressive, it may be true. I am also suspecting this is the root cause of test failure.